=== modified file 'scripts/ldd.pl'
--- old/scripts/ldd.pl	2010-08-26 20:35:15 +0000
+++ new/scripts/ldd.pl	2010-09-02 09:01:38 +0000
@@ -15,7 +15,7 @@
 my $Dynamic = 0;
 my $Format = '';
 
-my $Ldlinuxsodir = "/lib";
+my $Ldsodir = "/lib";
 my @Ld_Library_Path = qw(/usr/lib /lib /usr/lib32 /lib32 /usr/lib64 /lib64);
 
 
@@ -98,8 +98,8 @@
             next unless $line =~ /^ \s* NEEDED \s+ (.*) \s* $/x;
 
             my $needed = $1;
-            if ($needed =~ /^ld-linux(\.|-)/) {
-                $needed = "$Ldlinuxsodir/$needed";
+            if ($needed =~ /^ld(-linux)?(\.|-)/) {
+                $needed = "$Ldsodir/$needed";
             }
 
             ldso($needed);
@@ -182,7 +182,10 @@
         my $address = '0x' . '0' x ($Format =~ /^elf64-/ ? 16 : 8);
 
         foreach my $lib (@Libs) {
-            if (defined $Libs{$lib}) {
+            if ($lib =~ /^\//) {
+                printf "\t%s (%s)\n", $lib, $address;
+            }
+            elsif (defined $Libs{$lib}) {
                 printf "\t%s => %s (%s)\n", $lib, $Libs{$lib}, $address;
             }
             else {

