=== modified file 'src/fts.c'
--- a/src/fts.c	2010-12-18 00:58:00 +0000
+++ b/src/fts.c	2010-12-22 13:18:14 +0000
@@ -132,7 +132,7 @@
 	 * Start out with 1K of path space, and enough, in any case,
 	 * to hold the user's paths.
 	 */
-	if (fts_palloc(sp, MAX(fts_maxarglen(argv), MAXPATHLEN)))
+	if (fts_palloc(sp, MAX(fts_maxarglen(argv), FAKECHROOT_PATH_MAX)))
 		goto mem1;
 
 	/* Allocate/initialize root's parent. */

