=== modified file 'test/debootstrap.sh'
--- old/test/debootstrap.sh	2010-08-26 20:35:15 +0000
+++ new/test/debootstrap.sh	2010-09-02 09:01:38 +0000
@@ -9,7 +9,9 @@
 
 vendor=${VENDOR:-`lsb_release -s -i`}
 release=${RELEASE:-`lsb_release -s -c`}
-arch=${ARCH:-`dpkg-architecture -t$(arch)-linux-gnu -qDEB_HOST_ARCH 2>/dev/null`}
+type=`dpkg-architecture -qDEB_HOST_GNU_TYPE`
+systype=${type#*-}
+arch=${ARCH:-`dpkg-architecture -t$(arch)-$systype -qDEB_HOST_ARCH 2>/dev/null`}
 
 if [ $# -gt 0 ]; then
     destdir=$1

