From f53b566959fe252e87c7aa967a8c83405843d34b Mon Sep 17 00:00:00 2001
From: Piotr Roszatycki <piotr.roszatycki@gmail.com>
Date: Tue, 24 Dec 2013 18:52:42 +0100
Subject: [PATCH] Export V variable so it can be changed with ./configure

---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index d42ca10..279807f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,6 +10,8 @@ AM_INIT_AUTOMAKE([1.10 foreign])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AM_MAINTAINER_MODE
 
+AC_SUBST([V])
+
 LT_INIT
 
 AC_CANONICAL_HOST
-- 
1.8.3.2

